### PR TITLE
Add remaining irena data

### PR DESCRIPTION
**Pull request by pabloarosado**
Adapt snapshot, meadow, garden and grapher steps to new IRENA data, and update metadata.

### CODE REVIEW

File: snapshots/irena/2023-12-12/renewable_electricity_capacity_and_generation.xlsm.dvc [hunk 1]
```diff
@@ -1,48 +1,55 @@
...
-  date_accessed: 2023-06-28
-  is_public: true
-  description: |
-    IRENA publishes detailed statistics on renewable energy capacity, power generation and renewable energy balances. This data is collected directly from members using the IRENA Renewable Energy Statistics questionnaire and is also supplemented by desk research where official statistics are not available.
+      * Pumped storage is included under the "Hydropower" category but not in the "Total renewable energy". Generation from mixed plants is split between hydropower and pumped storage as appropriate.
 
-    Some technologies include others, following this schema:
+    # Citation
+    producer: International Renewable Energy Agency
+    citation_full: |-
```
Comment (veronikasamborska1994) [inline snapshots/irena/2023-12-12/renewable_electricity_capacity_and_generation.xlsm.dvc:35]: is there any reason to include all of this information in the citation full? the only recommended citation I can see is - `IRENA (2023), Renewable capacity statistics 2023, International Renewable Energy Agency, Abu Dhabi.`

File: snapshots/irena/2023-12-12/renewable_power_generation_costs.xlsx.dvc [hunk 2]
```diff
@@ -1,23 +1,31 @@
...
+  origin:
+    # Data product / Snapshot
+    title: Renewable Power Generation Costs
+    date_published: 2023-08-01
+    description: |
+
+    # Citation
+    producer: International Renewable Energy Agency
+    citation_full: |-
+      International Renewable Energy Agency (IRENA) - Renewable Power Generation Costs (2023).
```
Comment (veronikasamborska1994) [inline snapshots/irena/2023-12-12/renewable_power_generation_costs.xlsx.dvc:11]: and you don't seem to include all of the information regarding the copyright here

File: etl/steps/data/meadow/irena/2023-12-12/renewable_power_generation_costs.py [hunk 3]
```diff
@@ -63,36 +59,34 @@ def prepare_solar_pv_module_prices(excel_object: pr.ExcelFile) -> Table:
...
-    ).copy_metadata(_pv_prices)
+    )
 
     # Remove unnecessary column and add column for region.
-    pv_prices = pv_prices.drop(columns="technology").assign(**{"country": "World"})
+    pv_prices = pv_prices.drop(columns="technology", errors="raise").assign(**{"country": "World"})
 
     # Ignore years for which we don't have 12 months.
-    pv_prices = pv_prices[pv_prices["n_months"] == 12].drop(columns=["n_months"]).reset_index(drop=True)
+    pv_prices = pv_prices[pv_prices["n_months"] == 12].drop(columns=["n_months"], errors="raise").reset_index(drop=True)
```
Comment (veronikasamborska1994) [inline etl/steps/data/meadow/irena/2023-12-12/renewable_power_generation_costs.py:73]: I'd check that this works as expected (e.g. assertion error that only the most recent year is missing bc there is no data for 12 months; if there are other years for when that's the case it's good to add it to your metadata descriptions of the variables later on)

File: etl/steps/data/garden/irena/2023-12-12/renewable_power_generation_costs.countries.json [hunk 4]
```diff
@@ -14,10 +15,8 @@
     "Republic of Korea": "South Korea",
     "Spain": "Spain",
     "Sweden": "Sweden",
-    "Ukraine": "Ukraine",
     "United Kingdom": "United Kingdom",
     "United States": "United States",
-    "Viet Nam": "Vietnam",
```
Comment (veronikasamborska1994) [inline etl/steps/data/garden/irena/2023-12-12/renewable_power_generation_costs.countries.json:20]: do you know why the new version doesn't include the data for Vietnam and Ukraine?

File: etl/steps/data/garden/irena/2023-12-12/renewable_power_generation_costs.meta.yml [hunk 5]
```diff
@@ -1,46 +1,47 @@
...
+    processing_level: minor
+    presentation:
+      topic_tags:
+        - Energy
+    short_unit: $/kWh
+    unit: 2022 US$ per kilowatt-hour
+    description_short: |-
+      Measured in 2022 US$ per kilowatt-hour.
+
+  description_key_lcoe: &description_key_lcoe
```
Comment (veronikasamborska1994) [inline etl/steps/data/garden/irena/2023-12-12/renewable_power_generation_costs.meta.yml:12]: I'd include some of this in description_short as well.` "Average cost per unit of energy generated across the lifetime of a new power plant."`

File: etl/steps/data/garden/irena/2023-12-12/renewable_power_generation_costs.meta.yml [hunk 6]
```diff
@@ -1,46 +1,47 @@
...
 
 tables:
   renewable_power_generation_costs:
+    title: Renewable power generation costs
     variables:
       bioenergy:
         title: Bioenergy levelized cost of energy
-        short_unit: $/kWh
-        unit: 2021 US$ per kilowatt-hour
+        description_key: *description_key_lcoe
```
Comment (veronikasamborska1994) [inline etl/steps/data/garden/irena/2023-12-12/renewable_power_generation_costs.meta.yml:24]: Generally speaking I'd also add more specific descriptions in description key for what each of the technologies means more specifically (either take it from source or could use chatGPT to pre-generate them and then double check it all makes sense)

File: etl/steps/data/garden/irena/2023-12-12/renewable_power_generation_costs.meta.yml [hunk 7]
```diff
@@ -1,46 +1,47 @@
...
       onshore_wind:
         title: Onshore wind levelized cost of energy
-        short_unit: $/kWh
-        unit: 2021 US$ per kilowatt-hour
+        description_key: *description_key_lcoe
       solar_photovoltaic:
         title: Solar photovoltaic levelized cost of energy
```
Comment (veronikasamborska1994) [inline etl/steps/data/garden/irena/2023-12-12/renewable_power_generation_costs.meta.yml:41]: e.g., description_short/ description_key could be `Average total cost of generating electricity using solar panels over their entire lifespan, including installation, maintenance, and operational expenses, measured per unit of electricity produced.`

File: etl/steps/data/garden/irena/2023-12-12/renewable_power_generation_costs.meta.yml (hunk 5 above)
Comment (veronikasamborska1994) [inline etl/steps/data/garden/irena/2023-12-12/renewable_power_generation_costs.meta.yml:12]: on grapher it looks like the data was adjusted for inflation (I guess that's why it's in 2022 US dollars) - I'd include it in the description_key_lcoe as well

File: etl/steps/data/garden/irena/2023-12-12/renewable_electricity_capacity.meta.yml [hunk 8]
```diff
@@ -1,121 +1,59 @@
...
-        display:
-          numDecimalPlaces: 0
       liquid_biofuels:
         title: Liquid biofuels capacity
-        short_unit: MW
-        unit: megawatts
-        display:
-          numDecimalPlaces: 0
       marine_energy:
         title: Marine energy capacity
```
Comment (veronikasamborska1994) [inline etl/steps/data/garden/irena/2023-12-12/renewable_electricity_capacity.meta.yml:37]: it'd be useful to include similar description_key as you do for Bagasse for all/most of these. It's not immediately clear what `Marine energy capacity `or `Liquid biofuels capacity mean`.  Spelling out what capacity means would also be nice - ie. `Amount of electricity or energy that can be generated from x`.

File: etl/steps/data/garden/irena/2023-12-12/renewable_electricity_capacity.meta.yml [hunk 9]
```diff
@@ -1,121 +1,59 @@
...
+definitions:
+  common:
+    presentation:
+      topic_tags:
+        - Energy
+    short_unit: MW
+    unit: megawatts
+    display:
+      numDecimalPlaces: 0
+    description_short: |-
```
Comment (veronikasamborska1994) [inline etl/steps/data/garden/irena/2023-12-12/renewable_electricity_capacity.meta.yml:10]: that's where I'd add what you mean by capacity - eg., `Amount of electricity or energy that can be generated from x `

**Review by veronikasamborska1994 — COMMENTED** [review]
hey - I've taken a look and made a couple of suggestions mainly regarding making improvements to the metadata.

File: etl/steps/data/garden/irena/2023-12-12/renewable_power_generation_costs.countries.json (hunk 4 above)
Comment (pabloarosado) [reply etl/steps/data/garden/irena/2023-12-12/renewable_power_generation_costs.countries.json:20]: Good call, I think I've just figured it out: They added those (and a few others) as a separate sheet, for smaller markets... I'll try to combine them.

**Comment by pabloarosado** [comment]
Thanks a lot @veronikasamborska1994 I've added the missing data and improved some of the metadata as you suggested (https://github.com/owid/etl/pull/2116). I've written down some of the other changes as part of the energy bug bash, and I'll implement them only if there is time.

**Pull request merged**